### DISCUSSION TITLE
Todoist Integration

### DISCUSSION THREAD

**Discussion by agluck91** [makeplane/plane#6190] [Ideas]
I would love to see some integration with todoist. I do all of my planning and daily feed of tasks in todoist. It would be cool to be able to see plane tasks in my todoist daily view, and perhaps even generate a todo ticket from a ticket?

**Comment by danciaclara** [comment]
As mentioned in the other thread, you can use our APIs to implement a custom integration. I'm curious though—why would you need two different apps for tracking tasks? Is there something in Todoist that you find useful and isn’t available in Plane?

**Reply by agluck91** [reply]
Hey there. There is a very big difference for the use cases of the two platforms. 

For me, my entire day is planned in todoist. This is the function of a task manager. Todoist get's all of my calendar events and it let's me holistically see my day and what things are planned for that day. Then it also lets me bring tasks in and drop them on my timeline.  I have all my routines configured in it. Daily, monthly, yearly. I don't worry about renewing my license plate any more because when it is time to do that I will get a notification across all my devices. The interface removes all friction in that department. It does general task management very well. 
Something else that Todoist has that does not make any sense in Plane is conditional and location based notifications. Todoist will pop reminders up when I enter a certain geographical location. For instance, if I set a reminder to buy a specific thing, and that thing is at a store I don't go to frequently, I will just set the reminder to ding me when I get there. Then it is in the fore front of my mind before I ever walk in. This is also a very good use case of a general task manager.

Plane has a very different purpose, and it does that purpose EXTREMELY well. I do not see Plane as a task manager at all. I see it as a project planning platform. And frankly if you tried to become a general task manager it would ruin Plane, because it does project planning extremely well. I would never look to plane to manage my daily tasks. Even if you added all the bits and bobs that todoist had. It would make plane so chaotic and would be very hard to see relevant information. When I log into plane I know that I am going there to see how my intern is doing, see where we are at for my quarterly milestones, and will be picking up tasks that are relevant to my clients, I will be documenting the workflows and processes here that are relevant to a client's project. I have multiple clients that I service. So having a central location to separate, plan, and manage those clients is amazing.

And in the inverse, Todoist CAN do project planning it is not well suited for that function though. It is harder to collaborate on that platform that interface is not laid out well for that.  However, Todoist does task management exceptionally well. 

As for the integrations, I noticed that you all are already working on integrations. I will try to see what I can do, but do no have a lot of time to commit to it, so I posted it here to see if there was any appetite to bring integrations like this into the product development team. For example you all are already making a GitHub integration that I am eagerly waiting for. 
I have a few ideas about how I might go about implementing it and if I can make time I will start working on it. I definitely wanted to open the discussion though. I did not want to do this in a silo. I thought about a webhook that could automatically add a plane ticket to my Todoist inbox when the status is changed to 'In Progress'. That way I can see work that I am actively contributing to in my daily planning. Just bringing visibility to the tool where my whole day gets planned seems nice. I am still in the early stages of thinking this through though. In that workflow a by directional communication might be difficult but intersting. If a task that is tied to a plane ticket is closed in Todoist it could move  the state to done. You would have to default to a specific done state though. 

To be honest. I don't know if any of this makes sense to do though. I love the idea of it. Other planning platforms have integration todoist. Like trello and jira.
